### ISSUE TITLE
Question about abnormal response

### ISSUE THREAD

**Issue by Unrealluver** [haotian-liu/LLaVA#75]
Thanks for your awesome work again! When I run the pretrained code in the latest FSDP style code, I use the command here for a test.
```
python -m llava.eval.run_llava \
    --model-name /path/to/LLaVA-13B-v0 \
    --image-file "https://llava-vl.github.io/static/images/view.jpg" \
    --query "What are the things I should be cautious about when I visit here?"
```

But I got the output like this:
![f308bbde-8fc9-42ed-96ef-7e5992745043](https://user-images.githubusercontent.com/27504120/235353315-7d4cedda-8e62-49f7-a30b-f7b38d173fac.jpeg)

It seems that the model knows something about the image, but there are still some problems here that lead to a strange response.
Could you please help me with some possible reasons for that? Thank you very much!

**Comment by haotian-liu** [comment]
Hi @Unrealluver, I noticed a similar issue with the current implementation of FSDP in pretraining, however, the finetuned model behavior is normal.

I am suspecting that there is something unexpected happening with the writeback of the embeddings [here](https://github.com/haotian-liu/transformers_llava/blob/main/src/transformers/models/llama/modeling_llama.py#L581).

I am working on a refactoring of the framework, and it will be released very soon.  I think one temporary fix is as the other community member suggested: set the learning rate of all other modules to zero.

**Comment by haotian-liu** [comment]
Hi @Unrealluver, the issue has been fixed, and I trained for very few iterations (400 iterations) with LLaVA-7B checkpoint, and verified the results are reasonable.

We have also successfully moved the LLaVA implementation all into a single code base, so that we do not need to upgrade `transformers` package for new features.  Please upgrade to the latest code base and reinstall the `transformers` package, following the instruction below.  Thanks.

```
git pull
pip uninstall transformers
pip install git+https://github.com/huggingface/transformers@cae78c46
pip install -e .
```

**Comment by Unrealluver** [comment]
> Hi @Unrealluver, the issue has been fixed, and I trained for very few iterations (400 iterations) with LLaVA-7B checkpoint, and verified the results are reasonable.
> 
> We have also successfully moved the LLaVA implementation all into a single code base, so that we do not need to upgrade `transformers` package for new features. Please upgrade to the latest code base and reinstall the `transformers` package, following the instruction below. Thanks.
> 
> ```
> git pull
> pip uninstall transformers
> pip install git+https://github.com/huggingface/transformers@cae78c46
> pip install -e .
> ```

Thanks for your help, the latest commit truly solved the problem! But I found that the resume running will still face an error like bellow:

![image](https://user-images.githubusercontent.com/27504120/235464743-b01fc409-a914-41d3-b6aa-8a26e657dbb0.png)

Here my checkpoint iter is 2400, when I stop the pertaining and resume to run, the program returns the RuntimeError.

**Comment by haotian-liu** [comment]
Hi, thank you for pointing out this issue.. Since the pretraining on 8x A100s currently takes only around 2 hours (7B) and 4 hours (13B), I hope that it can be finished without interrupting.  I will investigate this issue later this week.  Thank you!

**Comment by Unrealluver** [comment]
Thanks for your reply.

**Comment by Unrealluver** [comment]
Hi, I am here to clarify the reason for the abnormal response. I noticed that in your latest code, we could get the true response, you switch the transformer library version to the version of the public repo and implement the llava model in `.llava/model/llava.py`. But I am still curious about the reason that leads to the abnormal response. Could you please give me more explanation about it? Thanks again!

**Comment by tingxueronghua** [comment]
> 

hello? have you found the reasons? I think the author tried to solve the problem by this commit: https://github.com/haotian-liu/LLaVA/commit/871bbd4a5248510156629e00ee5042b74f74764c

In this commit, I found the previous solution to keep the original embeddings parameters is to copy and paste it each time forwarding. And in this commit, he tries to block the gradients of corresponding embeddings by detach. I have no idea why this differs.
